src/Workspaces/Core/Portable/ExtractMethod/ExtractMethodOptions.cs

```diff
@@ -14,3 +22,16 @@ public ExtractMethodOptions()
 
     public static readonly ExtractMethodOptions Default = new();
 }
+
+internal readonly record struct ExtractMethodGenerationOptions(
```

Comment:
`ExtractMethodOptions` only contains options specific to method extraction. Additional options needed for code generation, naming and formatting are added in `ExtractMethodGenerationOptions`. Separating these options makes them composable to other option sets w/o duplication, e.g. `CodeActionOptions`.